.github/workflows/breaking-change-warning.yml

```diff
@@ -0,0 +1,50 @@
+name: Breaking Change Warning
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches:
+      - main
+      - release/*
+      - next
+
+jobs:
+  check_broken_types:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: get-files
+        run: |
+          echo "::set-output name=files::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | xargs)"
+
+      - name: Check for changes to broken types
+        id: check
+        run: |
+          declare -a arr=(${{ steps.get-files.outputs.files }})
+          for i in "${arr[@]}"; do
+            if [[ ! "$i" =~ package.json$ ]]; then
+              continue
+            fi
+
+            git checkout ${{ github.head_ref }} --quiet
+            pr=$(jq .typeValidation.broken "$i" | tr -d '\n')
+
+            git checkout ${{ github.base_ref }} --quiet
+            base=$(jq .typeValidation.broken "$i" | tr -d '\n')
+
+            if [ "$pr" != "$base" ]; then
+              echo '::set-output name=isBrokenDetected::true'
+              break
+            fi
+          done
+
+      - name: Comment on PR
+        if: steps.check.outputs.isBrokenDetected == 'true'
+        uses: marocchino/sticky-pull-request-comment@39c5b5dc7717447d0cba270cd115037d32d28443 # pin@v2
```

Comment:
A comment on why the version pinning might be helpful.

Reply:
I actually stole this action from: https://github.com/microsoft/FluidFramework/blob/18a4fd64371ea6925d99e99541c12e3b4adbed42/.github/workflows/push-queue.yml#L43

@sonalivdeshpande, can you speak to why we use the pin?

Reply:
We pin the actions mostly for security reasons: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

This is the action used for pinning workflows: https://github.com/mheap/pin-github-action